### PR TITLE
Issue 606 - Document clientside callbacks

### LINKED ISSUE #606: document clientside callbacks
Not something we necessarily *want* everyone to start using, but we have it so there should be docs.
https://github.com/plotly/dash-renderer/pull/143
https://github.com/plotly/dash/pull/672

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
Fixes #606 

Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [x] I understand

### CODE REVIEW

File: tutorial/examples/table/editing_loading_state.py [hunk 1]
```diff
@@ -8,6 +10,9 @@
 import dash_core_components as dcc
 import dash_html_components as html
 
+# For the documentation to always render the same values
+random.seed(0)
```
Comment (Marc-Andre-Rivet) [inline tutorial/examples/table/editing_loading_state.py]: @shammamah This random has some interesting effects when running percy diffs against this page 😁 

Seed the random so as to always output the same values. This prevents percy from identifying diffs on every run.

Comment (alexcjohnson) [reply tutorial/examples/table/editing_loading_state.py]: cc @wbrgss ^^

File: tutorial/performance.py [hunk 2]
```diff
@@ -109,6 +109,92 @@ def slow_function(input):
 your company would like to sponsor this work,
 [get in touch](https://plot.ly/products/consulting-and-oem/).
 
+'''),
+
+    dcc.Markdown('''
```
Comment (alexcjohnson) [inline tutorial/performance.py]: minor: could be combined with the previous `dcc.Markdown`?

File: tutorial/performance.py [hunk 3]
```diff
@@ -109,6 +109,92 @@ def slow_function(input):
...
+
+**Note**: It is important to keep in mind that clientside callbacks will
+execute on the browser's main thread and that they will block rendering
+and events processing while the being processed. Dash does not currently
+support asynchronous clientside callbacks and will fail if a `Promise`
+is returned.
+
+'''),
+
+    dcc.Markdown('''
```
Comment (alexcjohnson) [inline tutorial/performance.py]: same

File: tutorial/performance.py [hunk 4]
```diff
@@ -109,6 +109,92 @@ def slow_function(input):
...
+
+***
+
+### Clientside Callbacks
+
+Sometimes callbacks can incur a significant overhead, especially when they :
+- receive and/or return very large quantities of data (transfer time)
+- are called very often (queuing and handshake)
```
Comment (alexcjohnson) [inline tutorial/performance.py]: I'd specifically call out network latency here - this is what I've seen as the most common reason for using clientside, especially if the app is deployed in some restricted environment where there may be a lot of extra overhead due to VPNs, corporate monitoring, or whatever.

In fact I'd include some sort of a note about clientside callbacks not being possible when you need to refer to global variables on the server, or of course DB calls... which is very often the case when you have large quantities of data involved.

File: tutorial/performance.py (hunk 2 above)
Comment (Marc-Andre-Rivet) [reply tutorial/performance.py]: Sure - this was actually done voluntarily to make it clear they were different blocks but I don't mind.

Comment (Marc-Andre-Rivet) [reply tutorial/performance.py]: https://github.com/plotly/dash-docs/pull/655/commits/e3bdabf1ecddad38d2416694b51100f27b3d38ec

File: tutorial/examples/table/editing_loading_state.py [hunk 5]
```diff
@@ -1,4 +1,6 @@
 from time import sleep
+
+import random
 from random import randint
```
Comment (alexcjohnson) [inline tutorial/examples/table/editing_loading_state.py]: some linters will complain about `random` being imported 2 ways - so either just do `import random` or `from random import randint, seed`

File: tutorial/performance.py (hunk 2 above)
Comment (alexcjohnson) [reply tutorial/performance.py]: Yeah, understandable. But I think we want to err on the side of improved performance here; and markdown is pretty self-documenting about where the blocks break. Thanks for the change.

File: tutorial/performance.py (hunk 4 above)
Comment (Marc-Andre-Rivet) [reply tutorial/performance.py]: https://github.com/plotly/dash-docs/pull/655/commits/323c116438c130baa574bdd13b655cba8fa3099a

File: tutorial/examples/table/editing_loading_state.py (hunk 5 above)
Comment (Marc-Andre-Rivet) [reply tutorial/examples/table/editing_loading_state.py]: https://github.com/plotly/dash-docs/pull/655/commits/323c116438c130baa574bdd13b655cba8fa3099a

**Pull request merged**